target_game_3.0.py

```diff
@@ -0,0 +1,379 @@
+                    str(level) + ',' +
+                    str(mouse_click['mouse_x']) + ',' +
+                    str(mouse_click['mouse_y']) + ',' +
+                    str(mouse_click['reaction_time']) + ',' +
+                    str(mouse_click['mouse_in_target']) + '\n'
+                )
+
+
+# Here I create a function for the intro, which allows me to explain the game to the participant
+def show_intro(win):
+    start = TextStim(
+        win,
+        text = "Hi there, player! This is you! Let's play!",
+        height = 35,
+        color = (0.2,0.2,0.8),
+        pos = (250,230),
+    )
+    
+    player = ImageStim(
+        win,
+        'creature.jpg',
+        pos = (0,-100),
+        size=(420,420),
+    )
+    
+    intro_text = (
+        'In this game you should shoot at the target with your mouse. '
+        + 'The target moves rapidly to a new position each time you shoot at it.' 
+        + 'There are 3 levels each faster than the previous one. Good luck!'
+    )
+    
+    instructions = TextStim(
+        win,
+        text = intro_text,
+        height = 35,
+        color = (1, 0.2, 0.6),
+        pos = (250, 0),
+    )
+    
+    start.draw()
+    player.draw()
+    win.flip()
+    wait(3)
+    instructions.draw()
+    win.flip()
+    wait(8)
+
+# I set up my siplay size and background colour
+DISPSIZE = (1400, 800)
+BGC = (-1,-1,-1)
+
+DISPLAY_WIDTH = DISPSIZE[0]
+DISPLAY_HEIGHT = DISPSIZE[1]
+
+LIVES_X = -DISPLAY_WIDTH / 2
+LIVES_Y = DISPLAY_HEIGHT / 2
+
+
+# for my game I need to create some variables:
+score = 0
+lives =  3
+level = 1
+mouse_x = 0
+mouse_y = 0
+
+
+# I create some objects:
+win = Window(
+    size = DISPSIZE,
+    units = 'pix',
+    fullscr = False,
+    color = BGC
+    )
+
+mouse = Mouse(win)
+
+target = ImageStim(
+    win,
+    'target.png',
+    size = (420, 420)
+    )
+
+# I will display three text stimuli to the player while playing the game:
+lives_count = TextStim(
+    win,
+    text = f'Lives = {lives}',
+    height=35,
+    color = (1,0.2,0.6),
+    pos = (100, 330),
+    )
+
+score_count = TextStim(
+    win, 
+    text = f'Score = {score}', 
+    height = 35, 
+    color = (0.2,0.2,0.8), 
+    pos = (450,330)
+    )
+
+level_count = TextStim(
+    win, 
+    text = f'Level = {level}', 
+    height = 35,
+    color = (1, -0.5, 1), 
+    pos = (850, 330)
+    )
+
+# I define the messages to show the player the outcome of the game:
+you_have_lost = TextStim(
+    win, 
+    text = 'Boo! Not a great game, pal... Get it together!', 
+    height = 35, 
+    color = (0.2,0.2,0.8), 
+    pos = (250,230)
+    )
+
+you_have_won = TextStim(
+    win, 
+    text = 'Yey! Well done, champ! Time to celebrate!', 
+    height = 35, 
+    color = (0.2,0.2,0.8), 
+    pos = (250,230)
+    )
+
+# These are the images I use for the winning and loosing scenarios:
+looser = ImageStim(
+    win,
+    'failed.jpg',
+    pos = (0,-100),
+    size = (420,420)
+    )
+winner = ImageStim(

```

Comment:
Again, consistency. Usually you separate objects
```suggestion

winner = ImageStim(

```